python_modules/dagster-graphql/dagster_graphql/implementation/execution/__init__.py

```diff
@@ -143,6 +151,41 @@ def terminate_pipeline_execution(
     )
 
 
+def terminate_pipeline_execution_for_runs(
+    graphene_info: "ResolveInfo", run_ids: Sequence[str]
+) -> "GrapheneTerminateRunsResult":
+    from ...schema.roots.mutation import (
+        GrapheneTerminateRunPolicy,
+        GrapheneTerminateRunsResult,
+        GrapheneTerminateRunSuccess,
+    )
+
+    check.sequence_param(run_ids, "run_id", of_type=str)
+
+    run_ids_terminated = []
+    run_ids_failed_to_terminate = []
+
+    with DagsterInstance.from_ref(graphene_info.context.instance.get_ref()) as instance:
+        instance.optimize_for_webserver(DEFAULT_DB_STATEMENT_TIMEOUT, DEFAULT_POOL_RECYCLE)
+
+        for run_id in run_ids:
+            result = terminate_pipeline_execution(
+                graphene_info,
+                instance,
+                run_id,
+                cast(GrapheneTerminateRunPolicy, GrapheneTerminateRunPolicy.SAFE_TERMINATE),
```

Comment:
it seems to me like for this use case where you want lots of runs terminated ASAP we want the one that moves it into canceled immediately (GrapheneTerminateRunPolicy.MARK_AS_CANCELED_IMMEDIATELY)

or you could pass it in as an argument